File: src/EducationSystem/EducationSystem.WebApp/App/src/models/Discipline.ts
import Model from './Model'
import Test from './Test'
import Theme from './Theme'
import {IPagingOptions} from './PagedData'
import INameFilter from './Filters'

export default class Discipline extends Model {
  public Name: string = ''
  public Abbreviation: string = ''
  public Description: string = ''
  public Tests: Array<Test> = []
  public Themes: Array<Theme> = []
}

export interface IOptionsDiscipline {
  WithTests?: boolean,
  WithThemes?: boolean
}

export interface IFilterDiscipline extends IPagingOptions, INameFilter {}

File: src/EducationSystem/EducationSystem.Models/Rest/Discipline.cs
using System.Collections.Generic;
using EducationSystem.Models.Rest.Basics;

namespace EducationSystem.Models.Rest
{
    public class Discipline : Model
    {
        public string Name { get; set; }

        public string Abbreviation { get; set; }

        public string Description { get; set; }

        public List<Test> Tests { get; set; }

        public List<Theme> Themes { get; set; }

        public List<User> Lecturers { get; set; }
    }
}